java-vtl-script/src/test/java/no/ssb/vtl/script/functions/VTLSubstrTest.java

```diff
@@ -0,0 +1,165 @@
+package no.ssb.vtl.script.functions;
+
+/*
+ * ========================LICENSE_START=================================
+ * Java VTL
+ * %%
+ * Copyright (C) 2016 - 2017 Pawel Buczek
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import no.ssb.vtl.model.VTLNumber;
+import no.ssb.vtl.model.VTLObject;
+import no.ssb.vtl.model.VTLString;
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.*;
+
+
+public class VTLSubstrTest implements VTLNumberFunctionTest {
+
+    private VTLSubstr vtlSubstr;
+
+    @Before
+    public void setUp() throws Exception {
+        vtlSubstr = VTLSubstr.getInstance();
+    }
+
+    @Test
+    @Override
+    public void testInvokeWithPositiveNumber() throws Exception {
+        VTLObject<?> result = vtlSubstr.invoke(
+                createArguments("Hello, world!", 2, 5)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of("llo, "));
+
+        result = vtlSubstr.invoke(
+                createArguments("Hello, world!", 0, 4)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of("Hell"));
+
+        result = vtlSubstr.invoke(
+                createArguments("", 0, 4)
+        );
+        assertThat(result).isNotNull();
+        assertThat(result).isEqualTo(VTLString.of(""));
+
+        //deviation from the VTL specification 1.1: return empty string
+        //if start position greater than the whole length of the input string
+        result = vtlSubstr.invoke(
+                createArguments("Hello", 5, 2)
```

Comment:
`startPosition` begins with 0, so 5 is good :)
Ref. Part 2 line 2275
> B := substr(A, 0, 4) B = "Hell"

Reply:
`"Hello".substring(5)` returns `""`
`"Hello".substring(6)` throws an `StringIndexOutOfBoundsException`

Isn't the point of this test to ensure that any `startPosition` greater than the length of the input string will not fail and instead return `""`.
If someone removes the special logic handling the deviation then the test will still pass with a `startPosition` of `5`. However it will fail (as it should) if the test use `startPosition` `6`